webrender/src/device/gfx/device.rs

```diff
@@ -3431,20 +3435,54 @@ impl<B: hal::Backend> Device<B> {
                                 );
                             }
                             cmd_buffer.finish();
-                            true
                         }
-                        Err(_) => false,
+                        Err(acq_err) => {
+                            match acq_err {
+                                AcquireError::OutOfDate => {warn!("AcquireError : OutOfDate")},
+                                AcquireError::SurfaceLost(surf) => {warn!("AcquireError : SurfaceLost => {:?}", surf)},
+                                AcquireError::NotReady => {warn!("AcquireError : NotReady")},
+                                AcquireError::DeviceLost(dev) => {warn!("AcquireError : DeviceLost => {:?}", dev)},
+                                AcquireError::OutOfMemory(mem) => {warn!("AcquireError : OutOfMemory => {:?}", mem)}
+                            }
+                            self.wait_for_resize = true;
+                        },
                     }
                 }
                 None => {
                     self.current_frame_id = (self.current_frame_id + 1) % self.frame_count;
-                    true
                 }
             }
         }
     }
 
-    pub fn submit_to_gpu(&mut self) -> bool {
+    pub fn submit_to_gpu(&mut self) {
+        if self.wait_for_resize {
```

Comment:
I think we should move this block in a function since it's a duplication of the code form the end of the function.